### ISSUE TITLE
terragrunt can't be compiled from source without relying on GOPROXY

### ISSUE THREAD

**Issue by ekini** [gruntwork-io/terragrunt#2012]
The problem is still unnoticed because the Go module proxy caches packages without expiring them.

Terragrunt depends on old terraform version (due to using some of its functions), which depends on tencentcloud-sdk-go, which was inherently broken https://github.com/TencentCloud/tencentcloud-sdk-go/issues/125

The fix to terraform https://github.com/hashicorp/terraform/pull/29445/files can't be directly applied to terragrunt (maybe it 's still possible with replacements in go.mod?).

An onliner with Docker for a clean environment:
```bash
$docker run -ti --rm golang sh -c 'GOPROXY=direct go get github.com/gruntwork-io/terragrunt'
go: downloading github.com/gruntwork-io/terragrunt v0.36.1
go get: github.com/gruntwork-io/terragrunt@v0.36.1 requires
	github.com/hashicorp/terraform@v0.15.3 requires
	github.com/tencentcloud/tencentcloud-sdk-go@v3.0.82+incompatible: reading github.com/tencentcloud/tencentcloud-sdk-go/go.mod at revision v3.0.82: unknown revision v3.0.82
```

TLDR; the code will break at some point when the old lib gets purged from the go proxy cache.

More about this: https://drewdevault.com/2021/08/06/goproxy-breaks-go.html
